rspace-bench/src/test/scala/coop/rchain/rspace/bench/serialization/ModelSerializerBench.scala

```diff
@@ -0,0 +1,337 @@
+
+import coop.rchain.models._
+import coop.rchain.models.testImplicits._
+import coop.rchain.rspace.internal._
+import coop.rchain.rspace.Serialize
+import org.scalacheck.Gen.Parameters
+import org.scalacheck.rng.Seed
+import coop.rchain.rspace.bench.serialization._
+
+import org.openjdk.jmh.annotations.{State => BenchState, _}
+import org.openjdk.jmh.infra.Blackhole
+
+class ParSerializerBench {
+
+  import coop.rchain.rholang.interpreter.storage.implicits._
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 10)
+  @Measurement(iterations = 10)
+  @OperationsPerInvocation(value = 100)
+  def protoLight(bh: Blackhole, state: ProtobufModelBenchState) =
+    state.lightPars.foreach(e => bh.consume(state.roundTrip(e)))
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 10)
+  @Measurement(iterations = 10)
+  @OperationsPerInvocation(value = 100)
+  def protoMid(bh: Blackhole, state: ProtobufModelBenchState) =
+    state.midPars.foreach(e => bh.consume(state.roundTrip(e)))
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 10)
+  @Measurement(iterations = 10)
+  @OperationsPerInvocation(value = 100)
+  def protoHeavy(bh: Blackhole, state: ProtobufModelBenchState) =
+    state.heavyPars.foreach(e => bh.consume(state.roundTrip(e)))
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 10)
+  @Measurement(iterations = 10)
+  def kryoLight(bh: Blackhole, state: KryoModelBenchState) =
+    state.lightPars.foreach(e => bh.consume(state.parSerializer.roundTrip(e)))
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 10)
+  @Measurement(iterations = 10)
+  @OperationsPerInvocation(value = 100)
+  def kryoMid(bh: Blackhole, state: KryoModelBenchState) =
+    state.midPars.foreach(e => bh.consume(state.parSerializer.roundTrip(e)))
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 10)
+  @Measurement(iterations = 10)
+  @OperationsPerInvocation(value = 100)
+  def kryoHeavy(bh: Blackhole, state: KryoModelBenchState) =
+    state.heavyPars.foreach(e => bh.consume(state.parSerializer.roundTrip(e)))
+
+}
+
+class ExprSerializerBench {
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 10)
+  @Measurement(iterations = 10)
+  @OperationsPerInvocation(value = 100)
+  def protoLight(bh: Blackhole, state: ProtobufModelBenchState) =
+    state.lightExprs.foreach(e => bh.consume(state.roundTrip(e)(state.serializeExpr)))
```

Comment:
I don't think this is needed. Why do you think so?

Reply:
because it's overhead in the benchmark external to the tested code. Granted, its most likely negligible - but it's better to be sure than to measure that it indeed is negligible.


Don't insist though